### PR TITLE
NCAP-183 - Add access to GIT Webhooks and CMS VPN to Circle CI ELB, exclude open world

### PR DESCRIPTION
**Pull request by ghost**
What: We do not want to have CircleCI's ELB be open to 0.0.0.0./0, but rather just the CMS VPN IP addresses and the Git [Webhook origin addresses](https://help.github.com/en/articles/about-githubs-ip-addresses). This PR adds those IPs in a configurable way.

Testing: Test by bringing up a circleci stack and confirmed visually that the IPs were in the security group that was built by terraform and applied to the ELB.

image

### CODE REVIEW

**Comment by ghost** [comment]
The CCI terraform test is failing for the following reasons, I'm not familiar with this issue and don't think it is directly related to these code changes?

image

**Comment by msmorul** [comment]
No, those tests were included from the cloned repo from terraform. They should probably be disabled as they won't function in our environment. The reasoning for keeping was an attempt to minimize changes to the circle supplied items.

File: modules/cms/lb.tf
Comment (msmorul) [inline modules/cms/lb.tf:129]: I think this is safe, but am not sure if circle relies on any type of forwarding behavior

File: envs/dev-govcloud/main.tf
Comment (msmorul) [inline envs/dev-govcloud/main.tf]: for govcloud dev and prod, can you replace the github IP's with 52.203.194.136

File: envs/prod-govcloud/main.tf
Comment (msmorul) [inline envs/prod-govcloud/main.tf:20]: same ip change as govcloud-dev

Comment (ghost) [reply envs/prod-govcloud/main.tf:20]: fixed

**Review by msmorul — APPROVED** [review]
LGTM, please make sure this is pushed to the four environments on merge to master.

**Comment by ghost** [comment]
The Github IP link for future reference: https://api.github.com/meta

I'll hold this open waiting to hear back from Circle CI support to see if there is anything we are missing for whitelisting.

File: envs/dev/main.tf
Comment (cmonaghan) [inline envs/dev/main.tf]: It would be good to document where can these github webook IP's can be found. Perhaps add this link to the comment: https://api.github.com/meta

File: envs/prod/main.tf
Comment (cmonaghan) [inline envs/prod/main.tf]: same here

File: modules/cms/lb.tf
Comment (cmonaghan) [inline modules/cms/lb.tf]: Since the `"${var.ingress_ips}"` is a list, this may lead to a list within a list, like so: `[["52.203.194.136/32", "52.20.26.200/32", "34.196.35.156/32"]]`. Running `terraform plan` should tell you if this will work as currently defined. If not, then you can try just dropping the brackets here.

**Review by cmonaghan — APPROVED** [review]
LGTM. I'd also recommend running a `terraform apply` on at least one environment to verify it works as expected before merging the PR.

File: modules/cms/lb.tf
Comment (cmonaghan) [inline modules/cms/lb.tf]: this block should be retained, and set to allow traffic from the CMS VPN (otherwise we won't be able to access the circleci replicated console on port 8800)

File: modules/cms/lb.tf
Comment (ghost) [reply modules/cms/lb.tf]: I ran this yesterday on a fresh stack (terraform apply straight up) and it didn't error out and produced the security group as anticipated.

File: modules/cms/lb.tf
Comment (ghost) [reply modules/cms/lb.tf]: Got it, in that case we should probably separate GIT IPs from VPN IPs, will make that update shortly.

**Comment by ghost** [comment]
image

Testing the updated code, works exactly as intended

File: modules/cms/lb.tf
Comment (cmonaghan) [reply modules/cms/lb.tf]: oh cool. good to know

**Review by cmonaghan — APPROVED** [review]

**Pull request merged**
